packages/vaex-core/vaex/dataframe_protocol.py

```diff
@@ -0,0 +1,688 @@
+            n_chunks = self.num_chunks()
+            i = self._col.df.evaluate_iterator(self._col, chunk_size=size // n_chunks)
+            iterator = []
+            for i1, i2, chunk in i:
+                iterator.append(_VaexColumn(self._col[i1:i2]))
+            return iterator
+        elif self.num_chunks == 1:
+            size = self.size
+            i = self._col.df.evaluate_iterator(self._col, chunk_size=size // n_chunks)
+            iterator = []
+            for i1, i2, chunk in i:
+                iterator.append(_VaexColumn(self._col[i1:i2]))
+            return iterator
+
+        else:
+            raise ValueError(f"Column {self._col.expression} is already chunked.")
+
+    def get_buffers(self) -> Dict[str, Any]:
+        """
+        Return a dictionary containing the underlying buffers.
+
+        The returned dictionary has the following contents:
+
+            - "data": a two-element tuple whose first element is a buffer
+                      containing the data and whose second element is the data
+                      buffer's associated dtype.
+            - "validity": a two-element tuple whose first element is a buffer
+                          containing mask values indicating missing data and
+                          whose second element is the mask value buffer's
+                          associated dtype. None if the null representation is
+                          not a bit or byte mask.
+            - "offsets": a two-element tuple whose first element is a buffer
+                         containing the offset values for variable-size binary
+                         data (e.g., variable-length strings) and whose second
+                         element is the offsets buffer's associated dtype. None
+                         if the data buffer does not have an associated offsets
+                         buffer.
+        """
+        buffers = {}
+        buffers["data"] = self._get_data_buffer()
+        try:
+            buffers["validity"] = self._get_validity_buffer()
+        except:
+            buffers["validity"] = None
+
+        try:
+            buffers["offsets"] = self._get_offsets_buffer()
+        except:
+            buffers["offsets"] = None
+
+        return buffers
+
+    def _get_data_buffer(self) -> Tuple[_VaexBuffer, Any]:  # Any is for self.dtype tuple
+        """
+        Return the buffer containing the data and the buffer's associated dtype.
+        """
+        _k = _DtypeKind
+        if self.dtype[0] in (_k.INT, _k.UINT, _k.FLOAT, _k.BOOL):
+            # If arrow array is boolean .to_numpy changes values for some reason
+            # For that reason data is transferred to numpy through .tolist
+            if self.dtype[0] == _k.BOOL and isinstance(self._col.values, (pa.Array, pa.ChunkedArray)):
+                buffer = _VaexBuffer(np.array(self._col.tolist(), dtype=bool))
+            else:
+                buffer = _VaexBuffer(self._col.to_numpy())
```

Comment:
This could make a copy. We can use `.values/.evaluate()` but then we'd have to support arrow and numpy arrays, which should be fine, maybe the following works:
```suggestion
                buffer = _VaexBuffer(np.asarray(self._col.evaluate(), dtype=bool))
```